replications/remotewrite/writer.go

```diff
@@ -111,11 +111,7 @@ func (w *writer) Write(data []byte, attempts int) (backoff time.Duration, err er
 	}
 
 	res, postWriteErr := PostWrite(ctx, conf, data, w.clientTimeout)
-	res, msg, ok := normalizeResponse(res, postWriteErr)
-	if !ok {
-		// bail out
-		return w.backoff(attempts), postWriteErr
```

Comment:
I don't think we can't to lose this backoff calculation here. However, I do agree that we want to record the last return code/message. You can copy lines 121-128 into this if block so that we do update the response info, but also still attempt the backoff correctly.